src/Tasks/Unzip.cs

```diff
@@ -164,7 +164,11 @@ private void Extract(ZipArchive sourceArchive, DirectoryInfo destinationDirector
                     continue;
                 }
 
-                FileInfo destinationPath = new FileInfo(Path.Combine(destinationDirectory.FullName, zipArchiveEntry.FullName));
+                string fullDestinationPath = Path.GetFullPath(Path.Combine(destinationDirectory.FullName, zipArchiveEntry.FullName));
+                string fullDestinationDirectoryPath = Path.GetFullPath(destinationDirectory.FullName + Path.DirectorySeparatorChar);
```

Comment:
Can you pull this to outside the loop?

Reply:
The documentation seemed to want each ZipArchiveEntry checked individually, presumably because even if one entry is at a good path, the next could be at `..\..\..\...`, so I don't think so.

Reply:
Ah, yeah, good point. I can do that.